src/overlays/actors/ovl_En_Si/z_en_si.c

```diff
@@ -77,24 +88,80 @@ static DamageTable D_8098CDB8 = {
     /* Powder Keg     */ DMG_ENTRY(1, 0x0),
 };
 
-#endif
-
-extern ColliderSphereInit D_8098CD80;
-extern CollisionCheckInfoInit2 D_8098CDAC;
-extern DamageTable D_8098CDB8;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CA20.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CAD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CB70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CBDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Draw.s")
+void func_8098CA20(EnSi* this, PlayState* play) {
+    this->colliderSphere.dim.worldSphere.center.x = this->actor.world.pos.x;
+    this->colliderSphere.dim.worldSphere.center.y = this->actor.world.pos.y;
+    this->colliderSphere.dim.worldSphere.center.z = this->actor.world.pos.z;
+    this->colliderSphere.dim.worldSphere.radius =
+        this->colliderSphere.dim.modelSphere.radius * this->colliderSphere.dim.scale;
+    if (this->actor.colChkInfo.health > 0) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->colliderSphere.base);
+    }
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->colliderSphere.base);
+}
+
+void func_8098CAD0(EnSi* this, PlayState* play) {
+    s32 temp_a2 = ENSI_GET_PARAMS_FC(&this->actor);
+
+    if ((temp_a2 < 0x20) && (temp_a2 >= 0)) {
+        Flags_SetTreasure(play, temp_a2);
+    }
```

Comment:
```suggestion
    s32 chestFlag = ENSI_GET_CHEST_FLAG(&this->actor);

    if ((chestFlag < 0x20) && (chestFlag >= 0)) {
        Flags_SetTreasure(play, chestFlag);
    }
```
Just a suggestion, but I think this could be documented like this

Reply:
I won't pretend to understand this, but applied the suggestion nonetheless. Didn't know skulltula tokens could have a relation with chests (I only played OOT if that explains my ignorance).

Reply:
Yeah, that is actually pretty weird. The OoT equivalent of this file doesn't have this code in here. It was explicitly added in MM, but I can't think of any gold skulls in chests

Reply:
So I think it's just using the "chest flag" as the flag used to store whether a gold skull has been collected. In this case, it's not actually tied to a treasure chest, it's just using it's flag system

Reply:
Ok, interesting! Thanks for your help and explanations.